includes/options/views/class-amp-setup-wizard-submenu-page.php

```diff
@@ -0,0 +1,223 @@
+<?php
+/**
+ * AMP setup wizard page.
+ *
+ * @package AMP
+ * @since @todo NEW_ONBOARDING_RELEASE_VERSION
+ */
+
+/**
+ * AMP setup wizard submenu page class.
+ *
+ * @since @todo NEW_ONBOARDING_RELEASE_VERSION
+ */
+class AMP_Setup_Wizard_Submenu_Page {
+	/**
+	 * Handle for JS file.
+	 *
+	 * @var string
+	 */
+	const JS_HANDLE = 'amp-setup';
+
+	/**
+	 * HTML ID for the app root element.
+	 *
+	 * @var string
+	 */
+	const APP_ROOT_ID = 'amp-setup';
+
+	/**
+	 * The minimum core WP version whose core assets are compatibile with the setup feature.
+	 *
+	 * @var float
+	 */
+	const TARGET_WP_VERSION = 5.4;
+
+	/**
+	 * Sets up hooks.
+	 */
+	public function init() {
+		add_action( 'admin_enqueue_scripts', [ $this, 'override_scripts' ] );
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_assets' ] );
+	}
+
+	/**
+	 * Renders the setup screen markup.
+	 */
+	public function render() {
+		?>
+			<div id="<?php echo esc_attr( static::APP_ROOT_ID ); ?>"></div>
+		<?php
+	}
+
+	/**
+	 * Provides the setup screen handle.
+	 *
+	 * @return string
+	 */
+	public function screen_handle() {
+		return sprintf( 'amp_page_%s', AMP_Setup_Wizard_Submenu::SCREEN_ID );
+	}
+
+	/**
+	 * Registers single script.
+	 *
+	 * @param string  $handle  Script handle.
+	 * @param boolean $enqueue Whether to enqueue the asset immediately. Default false.
+	 */
+	public function add_setup_script( $handle, $enqueue = false ) {
+		$asset = $this->get_asset( $handle, sprintf( '%s/assets/js/%s.asset.php', AMP__DIR__, $handle ) );
+
+		wp_register_script(
+			$handle,
+			amp_get_asset_url( sprintf( 'js/%s.js', $handle ) ),
+			$asset['dependencies'],
+			$asset['version'],
+			true
+		);
+
+		if ( $enqueue ) {
+			wp_enqueue_script( $handle );
+		}
+	}
+
+	/**
+	 * Returns an asset's dependencies and version.
+	 *
+	 * @param string $handle Asset handle.
+	 * @param string $path   Path to file containing asset details.
+	 *
+	 * @return array Associative array containing the asset's dependencies and version.
+	 */
+	public function get_asset( $handle, $path = null ) {
+		/**
+		 * Filters AMP asset details.
+		 *
+		 * @param null|array $asset  Null or, to override, an array containing the asset's dependencies and version string.
+		 * @param string     $handle The asset handle.
+		 */
+		$asset = apply_filters( 'amp_setup_asset', null, $handle );
```

Comment:
I assume you're using the `null` with override here to avoid doing the file access when it would be overridden anyway?

Reply:
Yes, the immediate use case for this filter was in PHPUnit, where I'm using it across a few tests, so that the Webpack-generated files aren't required to exist. I also see it as potentially useful in future tasks where we may need a core package but none of its dependencies. It may may make sense at some point to change this filter to `amp_pre_setup_asset` and use `amp_setup_asset` to filter after the file is retrieved.